### PR TITLE
Escape CSV characters and save fulfilled orders to storage

### LINKED ISSUE #239: Fulfilling order does not update text file
Steps to recreate:
1. Create a purchase order
2. Fulfill the order
3. View the text file

image
image
image

### PR DESCRIPTION
**Pull request by yakultbottle**
Closes #239

### CODE REVIEW

File: src/main/java/seedu/pill/Pill.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ public final class Pill {
      */
     public void run() {
         items = storage.loadData();
-        transactionManager  = new TransactionManager(items);
+        transactionManager  = new TransactionManager(items, storage);
```
Comment (cnivedit) [inline src/main/java/seedu/pill/Pill.java:28]: Nice catch!

**Review by yijiano — APPROVED** [review]
LGTM, thanks

File: src/main/java/seedu/pill/util/Storage.java [hunk 2]
```diff
@@ -21,6 +23,14 @@ public class Storage {
...
     private static final String SEPARATOR = ",";
 
+    private static String escapeCommas(String input) {
+        return input.replace(",", "\\,");
+    }
+
+    private static String unescapeCommas(String input) {
+        return input.replace("\\,", ",");
+    }
+
```
Comment (cnivedit) [inline src/main/java/seedu/pill/util/Storage.java:33]: Well done with the alternative solution

File: src/main/java/seedu/pill/util/Parser.java [hunk 3]
```diff
@@ -575,9 +575,11 @@ && isValidDate(splitArguments[quantityIndex + 1])) {
             itemName = buildItemName(splitArguments, 0, splitArguments.length);
         }
 
+        /*
         if (itemName.contains(",")) {
             throw new PillException(ExceptionMessages.INVALID_ITEM_NAME);
         }
+        */
```
Comment (cnivedit) [inline src/main/java/seedu/pill/util/Parser.java:582]: Consider removing unused commented out code

**Review by cnivedit — APPROVED** [review]
LGTM

**Pull request merged**